src/shared/watchdog.c

```diff
@@ -115,8 +118,19 @@ int watchdog_set_timeout(usec_t *usec) {
 }
 
 int watchdog_ping(void) {
+        usec_t ntime;
         int r;
 
+        ntime = now(clock_boottime_or_monotonic());
+
+        if (watchdog_last_ping == USEC_INFINITY)
+                watchdog_last_ping = ntime;
```

Comment:
hmm, actually, why set this here? you set it again at the end of this function where it makes more sense

Reply:
Yes, I the proposed solution looks reasonable to me also. I don't know if the exported function name I used in the update is the most inspired one, but please comment and I'll do the changes in this patch. I'm also not sure if I should update the test-watchdog.c, not all watchdog interfaces are tested there. Definitely some tests will boost the confidence.